Ignore BOM when lexing

tobias-tengler opened the pull request:
If a GraphQL schema file should be encoded as `UTF-8 with BOM`, the lexer will skip the [BOM](https://en.wikipedia.org/wiki/Byte_order_mark) (`\ufeff`) character.

I've also added a regression snapshot test.

Closes #3798

captbaritone commented:
Rather than a snapshot test (where the BOM is very easy to overlook) could we create a unit test that uses an explicit escaped string?

tobias-tengler commented:
> Rather than a snapshot test (where the BOM is very easy to overlook) could we create a unit test that uses an explicit escaped string?

Done :)
I moved my test out of the schema document parsing tests and added it as a unit test directly to the lexer tests. I hope that's alright!
